### PR TITLE
KTLN-632: Check if a String Is a Palindrome in Kotlin

### PR DESCRIPTION
**Pull request by nappy29**
Added Unit test

### CODE REVIEW

File: core-kotlin-modules/core-kotlin-strings-3/src/test/kotlin/com/baeldung/palindromicstring/PalindromicStringTest.kt [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    }
+
+    fun isPalindromeUsingRecursion(str: String): Boolean {
+        if (str.length <= 1) {
+            return true
+        }
+        if (str.first() != str.last()) {
+            return false
+        }
+        return isPalindrome(str.substring(1, str.length - 1))
```
Comment (Bwvolleyball) [inline core-kotlin-modules/core-kotlin-strings-3/src/test/kotlin/com/baeldung/palindromicstring/PalindromicStringTest.kt:59]: Looks like this should be `isPalindromeUsingRecursion` instead.

```suggestion
        return isPalindromeUsingRecursion(str.substring(1, str.length - 1))
```

File: core-kotlin-modules/core-kotlin-strings-3/src/test/kotlin/com/baeldung/palindromicstring/PalindromicStringTest.kt [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        var j = str.length - 1
+        while (i < j) {
+            if (str[i] != str[j]) {
+                return false
+            }
+            i++
+            j--
+        }
+        return true
+    }
```
Comment (Bwvolleyball) [inline core-kotlin-modules/core-kotlin-strings-3/src/test/kotlin/com/baeldung/palindromicstring/PalindromicStringTest.kt:45]:
```suggestion
    fun isPalindrome(str: String): Boolean {
        var start = 0
        var end = str.length - 1
        while (start < end) {
            if (str[start] != str[end]) {
                return false
            }
            start++
            end--
        }
        return true
    }
```

Comment (nappy29) [reply core-kotlin-modules/core-kotlin-strings-3/src/test/kotlin/com/baeldung/palindromicstring/PalindromicStringTest.kt:45]: Got It Sir.

Comment (nappy29) [reply core-kotlin-modules/core-kotlin-strings-3/src/test/kotlin/com/baeldung/palindromicstring/PalindromicStringTest.kt:45]: Got It Sir.

File: core-kotlin-modules/core-kotlin-strings-3/src/test/kotlin/com/baeldung/palindromicstring/PalindromicStringTest.kt [hunk 3]
```diff
@@ -0,0 +1,60 @@
+package com.baeldung.palindromicstring
+import org.junit.jupiter.api.Test
+
+class PalindromicStringTest {
+
+    @Test fun `palindrome check using loop function`(){
+        assert(isPalindromeUsingLoop("racecar") == true)
```
Comment (KevinGilmore) [inline core-kotlin-modules/core-kotlin-strings-3/src/test/kotlin/com/baeldung/palindromicstring/PalindromicStringTest.kt:7]: For all trivial assertions, prefer either the JUnit5's assertTrue, assertFalse, etc, or use the AssertJ library's fluent assertions.

**Pull request merged**